atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/metrics/SnapshotTransactionMetricFactory.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.transaction.impl.metrics;
+
+import com.codahale.metrics.Counter;
+import com.codahale.metrics.Histogram;
+import com.codahale.metrics.Timer;
+import com.palantir.atlasdb.keyvalue.api.TableReference;
+import com.palantir.atlasdb.transaction.impl.SnapshotTransaction;
+import com.palantir.atlasdb.util.MetricsManager;
+
+// We continue to use the SnapshotTransaction origin to avoid a breaking change in our metric names.
+public final class SnapshotTransactionMetricFactory {
+    private final MetricsManager metricsManager;
+    private final TableLevelMetricsController tableLevelMetricsController;
+
+    public SnapshotTransactionMetricFactory(
+            MetricsManager metricsManager, TableLevelMetricsController tableLevelMetricsController) {
+        this.metricsManager = metricsManager;
+        this.tableLevelMetricsController = tableLevelMetricsController;
+    }
+
+    public Timer getTimer(String name) {
+        return metricsManager.registerOrGetTimer(SnapshotTransaction.class, name);
```

Comment:
How do you feel about having a `private final Class<SnapshotTransaction> delegateOrigin = SnapshotTransaction.class;` and just reusing it across the class?

So here would become:

```
 return metricsManager.registerOrGetTimer(delegateOrigin, name);
```

Reply:
yeah, makes sense 👍

Reply:
(I'll probably make it static unless there's a reason not to?)